ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/diffbased/common/worldview/WorldStateStorageConfig.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright contributors to Besu.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.trie.diffbased.common.worldview;
+
+public class WorldStateStorageConfig {
+
+  private final boolean isFrozen;
+
+  private WorldStateStorageConfig(final Builder builder) {
+    this.isFrozen = builder.isFrozen;
+  }
+
+  public boolean isFrozen() {
+    return isFrozen;
+  }
+
+  public static WorldStateStorageConfig.Builder newBuilder() {
+    return new WorldStateStorageConfig.Builder();
+  }
```

Comment:
A class with a single boolean member var (for now) having a builder seems like overkill IMO.  A single static constructor would probably be sufficient.  Will Verkle or stateless extend this class with its own configs?

Reply:
yes I think it will be needed for stateless. I can remove it later if I don't use it in the future

Reply:
removed the class